opentasks/src/main/java/org/dmfs/tasks/utils/RecentlyUsedLists.java

```diff
@@ -39,19 +47,24 @@
      *
      * @return List of TaskLists where the most recently used list is on position 0.
      */
-    public static List<Long> getList(Context context)
+    private static Iterable<Long> getList(Context context)
     {
-        String strLists = PreferenceManager.getDefaultSharedPreferences(context).getString(PREFERENCE_KEY, "");
-        Log.v(RecentlyUsedLists.class.getSimpleName(), "getList:  " + strLists);
-        List<Long> lists = new ArrayList<>();
-        if (strLists.length() > 0)
+        Optional<String> strLists = new NullSafe<>(PreferenceManager.getDefaultSharedPreferences(context)
+                .getString(PREFERENCE_KEY, null));
+
+        Log.v(RecentlyUsedLists.class.getSimpleName(), "getList:  " + strLists.value("empty"));
+
+        if (!strLists.isPresent())
         {
-            for (String lid : strLists.split(","))
-            {
-                lists.add(Long.parseLong(lid));
-            }
+            return new EmptyIterable<>();
         }
-        return lists;
+
+        return new Mapped<>(cs -> Long.valueOf(cs.toString()),
+                new Filtered<>(
+                        new Split(strLists.value(), ','),
+                        // 'null' should not be here but happened before: https://github.com/dmfs/opentasks/issues/562
+                        // TODO Checking for null can be removed after a while since stored values will be replaced with correct values on people's devices with this update
+                        Objects::nonNull));
```

Comment:
This check doesn't make any sense. `Split` will not iterate `null`. You need to check for the String `"null"`.
Also the original issue wasn't about a `null` reference, but about `"null"` not being the string representation of a long value.

Reply:
Oh, that was clearly mistaken. Fixed it.